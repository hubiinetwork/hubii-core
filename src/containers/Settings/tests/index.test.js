// import React from 'react';
// import { shallow } from 'enzyme';

// import { Settings } from '../index';

describe('<Settings />', () => {
  it('Expect to have unit tests specified', () => {
    expect(true).toEqual(false);
  });
});
