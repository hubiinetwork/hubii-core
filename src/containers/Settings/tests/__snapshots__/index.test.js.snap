// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Settings
    currentNetwork={
        Object {
            "identityServiceAppId": "appid",
            "identityServiceSecret": "secret",
            "nahmiiProvider": NahmiiProvider {
              "_balances": Object {},
              "_emitted": Object {
                "block": -2,
              },
              "_events": Array [],
              "_fastQueryDate": 0,
              "_lastBlockNumber": -2,
              "_network": Object {
                "_defaultProvider": [Function],
                "chainId": 3,
                "ensAddress": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
                "name": "ropsten",
              },
              "_pollingInterval": 4000,
              "connection": Object {
                "url": "http://geth-ropsten.dev.hubii.net",
              },
              "ready": Promise {},
            },
            "provider": FallbackProvider {
              "_balances": Object {},
              "_emitted": Object {
                "block": -2,
              },
              "_events": Array [],
              "_fastQueryDate": 0,
              "_lastBlockNumber": -2,
              "_network": Object {
                "_defaultProvider": [Function],
                "chainId": 3,
                "ensAddress": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
                "name": "ropsten",
              },
              "_pollingInterval": 4000,
              "_providers": Array [
                InfuraProvider {
                  "_balances": Object {},
                  "_emitted": Object {
                    "block": -2,
                  },
                  "_events": Array [],
                  "_fastQueryDate": 0,
                  "_lastBlockNumber": -2,
                  "_network": Object {
                    "_defaultProvider": [Function],
                    "chainId": 3,
                    "ensAddress": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
                    "name": "ropsten",
                  },
                  "_pollingInterval": 4000,
                  "apiAccessToken": null,
                  "connection": Object {
                    "url": "https://ropsten.infura.io/",
                  },
                  "ready": Promise {},
                },
                EtherscanProvider {
                  "_balances": Object {},
                  "_emitted": Object {
                    "block": -2,
                  },
                  "_events": Array [],
                  "_fastQueryDate": 0,
                  "_lastBlockNumber": -2,
                  "_network": Object {
                    "_defaultProvider": [Function],
                    "chainId": 3,
                    "ensAddress": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
                    "name": "ropsten",
                  },
                  "_pollingInterval": 4000,
                  "apiKey": undefined,
                  "baseUrl": "https://api-ropsten.etherscan.io",
                  "ready": Promise {},
                },
              ],
              "ready": Promise {},
            },
            "walletApiEndpoint": [Function],
          }
    }
    intl={
        Object {
            "formatMessage": [Function],
          }
    }
    locale="en"
    onChangeLocale={[Function]}
    onChangeNetwork={[Function]}
    supportedNetworks={
        Immutable.Map {
            homestead: Immutable.Map {
                provider: FallbackProvider {
                    "_balances": Object {},
                    "_emitted": Object {
                      "block": -2,
                    },
                    "_events": Array [],
                    "_fastQueryDate": 0,
                    "_lastBlockNumber": -2,
                    "_network": Object {
                      "_defaultProvider": [Function],
                      "chainId": 1,
                      "ensAddress": "0x314159265dd8dbb310642f98f50c066173c1259b",
                      "name": "homestead",
                    },
                    "_pollingInterval": 4000,
                    "_providers": Array [
                      InfuraProvider {
                        "_balances": Object {},
                        "_emitted": Object {
                          "block": -2,
                        },
                        "_events": Array [],
                        "_fastQueryDate": 0,
                        "_lastBlockNumber": -2,
                        "_network": Object {
                          "_defaultProvider": [Function],
                          "chainId": 1,
                          "ensAddress": "0x314159265dd8dbb310642f98f50c066173c1259b",
                          "name": "homestead",
                        },
                        "_pollingInterval": 4000,
                        "apiAccessToken": null,
                        "connection": Object {
                          "url": "https://mainnet.infura.io/",
                        },
                        "ready": Promise {},
                      },
                      EtherscanProvider {
                        "_balances": Object {},
                        "_emitted": Object {
                          "block": -2,
                        },
                        "_events": Array [],
                        "_fastQueryDate": 0,
                        "_lastBlockNumber": -2,
                        "_network": Object {
                          "_defaultProvider": [Function],
                          "chainId": 1,
                          "ensAddress": "0x314159265dd8dbb310642f98f50c066173c1259b",
                          "name": "homestead",
                        },
                        "_pollingInterval": 4000,
                        "apiKey": undefined,
                        "baseUrl": "https://api.etherscan.io",
                        "ready": Promise {},
                      },
                    ],
                    "ready": Promise {},
                  },
                nahmiiProvider: NahmiiProvider {
                    "_balances": Object {},
                    "_emitted": Object {
                      "block": -2,
                    },
                    "_events": Array [],
                    "_fastQueryDate": 0,
                    "_lastBlockNumber": -2,
                    "_network": Object {
                      "_defaultProvider": [Function],
                      "chainId": 3,
                      "ensAddress": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
                      "name": "ropsten",
                    },
                    "_pollingInterval": 4000,
                    "connection": Object {
                      "url": "http://geth-ropsten.dev.hubii.net",
                    },
                    "ready": Promise {},
                  },
                walletApiEndpoint: [Function],
                identityServiceSecret: "secret",
                identityServiceAppId: "appid",
            },
            ropsten: Immutable.Map {
                provider: FallbackProvider {
                    "_balances": Object {},
                    "_emitted": Object {
                      "block": -2,
                    },
                    "_events": Array [],
                    "_fastQueryDate": 0,
                    "_lastBlockNumber": -2,
                    "_network": Object {
                      "_defaultProvider": [Function],
                      "chainId": 3,
                      "ensAddress": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
                      "name": "ropsten",
                    },
                    "_pollingInterval": 4000,
                    "_providers": Array [
                      InfuraProvider {
                        "_balances": Object {},
                        "_emitted": Object {
                          "block": -2,
                        },
                        "_events": Array [],
                        "_fastQueryDate": 0,
                        "_lastBlockNumber": -2,
                        "_network": Object {
                          "_defaultProvider": [Function],
                          "chainId": 3,
                          "ensAddress": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
                          "name": "ropsten",
                        },
                        "_pollingInterval": 4000,
                        "apiAccessToken": null,
                        "connection": Object {
                          "url": "https://ropsten.infura.io/",
                        },
                        "ready": Promise {},
                      },
                      EtherscanProvider {
                        "_balances": Object {},
                        "_emitted": Object {
                          "block": -2,
                        },
                        "_events": Array [],
                        "_fastQueryDate": 0,
                        "_lastBlockNumber": -2,
                        "_network": Object {
                          "_defaultProvider": [Function],
                          "chainId": 3,
                          "ensAddress": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
                          "name": "ropsten",
                        },
                        "_pollingInterval": 4000,
                        "apiKey": undefined,
                        "baseUrl": "https://api-ropsten.etherscan.io",
                        "ready": Promise {},
                      },
                    ],
                    "ready": Promise {},
                  },
                nahmiiProvider: NahmiiProvider {
                    "_balances": Object {},
                    "_emitted": Object {
                      "block": -2,
                    },
                    "_events": Array [],
                    "_fastQueryDate": 0,
                    "_lastBlockNumber": -2,
                    "_network": Object {
                      "_defaultProvider": [Function],
                      "chainId": 3,
                      "ensAddress": "0x112234455c3a32fd11230c42e7bccd4a84e02010",
                      "name": "ropsten",
                    },
                    "_pollingInterval": 4000,
                    "connection": Object {
                      "url": "http://geth-ropsten.dev.hubii.net",
                    },
                    "ready": Promise {},
                  },
                walletApiEndpoint: [Function],
                identityServiceSecret: "secret",
                identityServiceAppId: "appid",
            },
        }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TopHeader>
          <Heading>
                    settings
          </Heading>
</TopHeader>,
        <style__Body>
          <style__SettingWrapper>
                    <style__StyledSectionHeading>
                              setting_network
                    </style__StyledSectionHeading>
                    <Select.style
                              onChange={[Function]}
                              style={
                                        Object {
                                                  "width": "15rem",
                                                }
                              }
                              value="ropsten"
                    >
                              <Option
                                        value="homestead"
                              >
                                        Homestead [MAINNET]
                              </Option>
                              <Option
                                        value="ropsten"
                              >
                                        Ropsten [TESTNET]
                              </Option>
                    </Select.style>
          </style__SettingWrapper>
          <style__SettingWrapper>
                    <style__StyledSectionHeading>
                              setting_language
                    </style__StyledSectionHeading>
                    <Select.style
                              onChange={[Function]}
                              style={
                                        Object {
                                                  "width": "15rem",
                                                }
                              }
                              value="en"
                    >
                              <Option
                                        value="en"
                              >
                                        English
                              </Option>
                    </Select.style>
          </style__SettingWrapper>
</style__Body>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Heading>
            settings
</Heading>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "settings",
          },
          "ref": null,
          "rendered": "settings",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <style__SettingWrapper>
              <style__StyledSectionHeading>
                            setting_network
              </style__StyledSectionHeading>
              <Select.style
                            onChange={[Function]}
                            style={
                                          Object {
                                                        "width": "15rem",
                                                      }
                            }
                            value="ropsten"
              >
                            <Option
                                          value="homestead"
                            >
                                          Homestead [MAINNET]
                            </Option>
                            <Option
                                          value="ropsten"
                            >
                                          Ropsten [TESTNET]
                            </Option>
              </Select.style>
</style__SettingWrapper>,
            <style__SettingWrapper>
              <style__StyledSectionHeading>
                            setting_language
              </style__StyledSectionHeading>
              <Select.style
                            onChange={[Function]}
                            style={
                                          Object {
                                                        "width": "15rem",
                                                      }
                            }
                            value="en"
              >
                            <Option
                                          value="en"
                            >
                                          English
                            </Option>
              </Select.style>
</style__SettingWrapper>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <style__StyledSectionHeading>
                  setting_network
</style__StyledSectionHeading>,
                <Select.style
                  onChange={[Function]}
                  style={
                                    Object {
                                                      "width": "15rem",
                                                    }
                  }
                  value="ropsten"
>
                  <Option
                                    value="homestead"
                  >
                                    Homestead [MAINNET]
                  </Option>
                  <Option
                                    value="ropsten"
                  >
                                    Ropsten [TESTNET]
                  </Option>
</Select.style>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "setting_network",
                },
                "ref": null,
                "rendered": "setting_network",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Option
                      value="homestead"
>
                      Homestead [MAINNET]
</Option>,
                    <Option
                      value="ropsten"
>
                      Ropsten [TESTNET]
</Option>,
                  ],
                  "onChange": [Function],
                  "style": Object {
                    "width": "15rem",
                  },
                  "value": "ropsten",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "homestead",
                    "nodeType": "class",
                    "props": Object {
                      "children": "Homestead [MAINNET]",
                      "value": "homestead",
                    },
                    "ref": null,
                    "rendered": "Homestead [MAINNET]",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": "ropsten",
                    "nodeType": "class",
                    "props": Object {
                      "children": "Ropsten [TESTNET]",
                      "value": "ropsten",
                    },
                    "ref": null,
                    "rendered": "Ropsten [TESTNET]",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <style__StyledSectionHeading>
                  setting_language
</style__StyledSectionHeading>,
                <Select.style
                  onChange={[Function]}
                  style={
                                    Object {
                                                      "width": "15rem",
                                                    }
                  }
                  value="en"
>
                  <Option
                                    value="en"
                  >
                                    English
                  </Option>
</Select.style>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "setting_language",
                },
                "ref": null,
                "rendered": "setting_language",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Option
                      value="en"
>
                      English
</Option>,
                  ],
                  "onChange": [Function],
                  "style": Object {
                    "width": "15rem",
                  },
                  "value": "en",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "en",
                    "nodeType": "class",
                    "props": Object {
                      "children": "English",
                      "value": "en",
                    },
                    "ref": null,
                    "rendered": "English",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TopHeader>
            <Heading>
                        settings
            </Heading>
</TopHeader>,
          <style__Body>
            <style__SettingWrapper>
                        <style__StyledSectionHeading>
                                    setting_network
                        </style__StyledSectionHeading>
                        <Select.style
                                    onChange={[Function]}
                                    style={
                                                Object {
                                                            "width": "15rem",
                                                          }
                                    }
                                    value="ropsten"
                        >
                                    <Option
                                                value="homestead"
                                    >
                                                Homestead [MAINNET]
                                    </Option>
                                    <Option
                                                value="ropsten"
                                    >
                                                Ropsten [TESTNET]
                                    </Option>
                        </Select.style>
            </style__SettingWrapper>
            <style__SettingWrapper>
                        <style__StyledSectionHeading>
                                    setting_language
                        </style__StyledSectionHeading>
                        <Select.style
                                    onChange={[Function]}
                                    style={
                                                Object {
                                                            "width": "15rem",
                                                          }
                                    }
                                    value="en"
                        >
                                    <Option
                                                value="en"
                                    >
                                                English
                                    </Option>
                        </Select.style>
            </style__SettingWrapper>
</style__Body>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Heading>
              settings
</Heading>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "settings",
            },
            "ref": null,
            "rendered": "settings",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <style__SettingWrapper>
                <style__StyledSectionHeading>
                                setting_network
                </style__StyledSectionHeading>
                <Select.style
                                onChange={[Function]}
                                style={
                                                Object {
                                                                "width": "15rem",
                                                              }
                                }
                                value="ropsten"
                >
                                <Option
                                                value="homestead"
                                >
                                                Homestead [MAINNET]
                                </Option>
                                <Option
                                                value="ropsten"
                                >
                                                Ropsten [TESTNET]
                                </Option>
                </Select.style>
</style__SettingWrapper>,
              <style__SettingWrapper>
                <style__StyledSectionHeading>
                                setting_language
                </style__StyledSectionHeading>
                <Select.style
                                onChange={[Function]}
                                style={
                                                Object {
                                                                "width": "15rem",
                                                              }
                                }
                                value="en"
                >
                                <Option
                                                value="en"
                                >
                                                English
                                </Option>
                </Select.style>
</style__SettingWrapper>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <style__StyledSectionHeading>
                    setting_network
</style__StyledSectionHeading>,
                  <Select.style
                    onChange={[Function]}
                    style={
                                        Object {
                                                            "width": "15rem",
                                                          }
                    }
                    value="ropsten"
>
                    <Option
                                        value="homestead"
                    >
                                        Homestead [MAINNET]
                    </Option>
                    <Option
                                        value="ropsten"
                    >
                                        Ropsten [TESTNET]
                    </Option>
</Select.style>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "setting_network",
                  },
                  "ref": null,
                  "rendered": "setting_network",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <Option
                        value="homestead"
>
                        Homestead [MAINNET]
</Option>,
                      <Option
                        value="ropsten"
>
                        Ropsten [TESTNET]
</Option>,
                    ],
                    "onChange": [Function],
                    "style": Object {
                      "width": "15rem",
                    },
                    "value": "ropsten",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "homestead",
                      "nodeType": "class",
                      "props": Object {
                        "children": "Homestead [MAINNET]",
                        "value": "homestead",
                      },
                      "ref": null,
                      "rendered": "Homestead [MAINNET]",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": "ropsten",
                      "nodeType": "class",
                      "props": Object {
                        "children": "Ropsten [TESTNET]",
                        "value": "ropsten",
                      },
                      "ref": null,
                      "rendered": "Ropsten [TESTNET]",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <style__StyledSectionHeading>
                    setting_language
</style__StyledSectionHeading>,
                  <Select.style
                    onChange={[Function]}
                    style={
                                        Object {
                                                            "width": "15rem",
                                                          }
                    }
                    value="en"
>
                    <Option
                                        value="en"
                    >
                                        English
                    </Option>
</Select.style>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "setting_language",
                  },
                  "ref": null,
                  "rendered": "setting_language",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <Option
                        value="en"
>
                        English
</Option>,
                    ],
                    "onChange": [Function],
                    "style": Object {
                      "width": "15rem",
                    },
                    "value": "en",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "en",
                      "nodeType": "class",
                      "props": Object {
                        "children": "English",
                        "value": "en",
                      },
                      "ref": null,
                      "rendered": "English",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
